src/signTxPoolRegistration.c

```diff
@@ -0,0 +1,836 @@
+			uint8_t vrfKeyHash[VRF_KEY_HASH_LENGTH];
+			uint8_t pledge[8];
+			uint8_t cost[8];
+			uint8_t marginNumerator[8];
+			uint8_t marginDenominator[8];
+			uint8_t rewardAccount[1 + ADDRESS_KEY_HASH_LENGTH];
+
+			uint8_t numOwners[4];
+			uint8_t numRelays[4];
+		}* wireHeader = (void*) wireDataBuffer;
+
+		TRACE("%d %d", SIZEOF(*wireHeader), wireDataSize);
+		VALIDATE(SIZEOF(*wireHeader) == wireDataSize, ERR_INVALID_DATA);
+
+		{
+			pool_registration_params_t* p = &subctx->poolParams;
+
+			TRACE_BUFFER(wireHeader->operatorHash, SIZEOF(wireHeader->operatorHash));
+			ASSERT(SIZEOF(wireHeader->operatorHash) == SIZEOF(p->operatorHash));
+			os_memmove(p->operatorHash, wireHeader->operatorHash, SIZEOF(p->operatorHash));
+			// nothing to validate, all values are valid
+
+			TRACE_BUFFER(wireHeader->vrfKeyHash, SIZEOF(wireHeader->vrfKeyHash));
+			ASSERT(SIZEOF(wireHeader->vrfKeyHash) == SIZEOF(p->vrfKeyHash));
+			os_memmove(p->vrfKeyHash, wireHeader->vrfKeyHash, SIZEOF(p->vrfKeyHash));
+			// nothing to validate, all values are valid
+
+			ASSERT_TYPE(p->pledge, uint64_t);
+			p->pledge = u8be_read(wireHeader->pledge);
+			TRACE_ADA_AMOUNT("pledge ", p->pledge);
+			VALIDATE(p->pledge < LOVELACE_MAX_SUPPLY, ERR_INVALID_DATA);
+
+			ASSERT_TYPE(p->cost, uint64_t);
+			p->cost = u8be_read(wireHeader->cost);
+			TRACE_ADA_AMOUNT("cost ", p->cost);
+			VALIDATE(p->cost < LOVELACE_MAX_SUPPLY, ERR_INVALID_DATA);
+
+			ASSERT_TYPE(p->marginNumerator, uint64_t);
+			p->marginNumerator = u8be_read(wireHeader->marginNumerator);
+			TRACE_BUFFER((uint8_t *) &p->marginNumerator, 8);
+			VALIDATE(p->marginNumerator <= MARGIN_DENOMINATOR_MAX, ERR_INVALID_DATA);
+
+			ASSERT_TYPE(p->marginDenominator, uint64_t);
+			p->marginDenominator = u8be_read(wireHeader->marginDenominator);
+			TRACE_BUFFER((uint8_t *) &p->marginDenominator, 8);
+			VALIDATE(p->marginDenominator != 0, ERR_INVALID_DATA);
+			VALIDATE(p->marginDenominator <= MARGIN_DENOMINATOR_MAX, ERR_INVALID_DATA);
+			VALIDATE(p->marginNumerator <= p->marginDenominator, ERR_INVALID_DATA);
+
+			TRACE_BUFFER(wireHeader->rewardAccount, SIZEOF(wireHeader->rewardAccount));
+			ASSERT(SIZEOF(wireHeader->rewardAccount) == SIZEOF(p->rewardAccount));
+			os_memmove(p->rewardAccount, wireHeader->rewardAccount, SIZEOF(p->rewardAccount));
+			const uint8_t header = getAddressHeader(p->rewardAccount, SIZEOF(p->rewardAccount));
+			VALIDATE(getAddressType(header) == REWARD, ERR_INVALID_DATA);
+			VALIDATE(getNetworkId(header) == commonTxData->networkId, ERR_INVALID_DATA);
+		}
+
+		ASSERT_TYPE(subctx->numOwners, uint16_t);
+		ASSERT_TYPE(subctx->numRelays, uint16_t);
+		subctx->numOwners = (uint16_t) u4be_read(wireHeader->numOwners);
+		subctx->numRelays = (uint16_t) u4be_read(wireHeader->numRelays);
+
+		TRACE(
+		        "num owners, relays: %d %d",
+		        subctx->numOwners, subctx->numRelays
+		);
+		VALIDATE(subctx->numOwners < POOL_MAX_OWNERS, ERR_INVALID_DATA);
```

Comment:
shouldn't this technically be a `<=` comparison? I mean if something is declared to be the "MAX" value, I would expect the max itself to be a valid value

Reply:
Yes, though this is the way how it was done previously for inputs etc. I have changed it in other places too.